include/exadg/solvers_and_preconditioners/newton/newton_solver.h

```diff
@@ -80,12 +80,11 @@ class Solver
       linear_operator.set_solution_linearization(solution);
 
       // determine whether to update the operator/preconditioner of the linearized problem
-      bool const threshold_exceeded = (newton_iterations % update.threshold_newton_iter == 0) ||
-                                      (linear_iterations_last > update.threshold_linear_iter);
+      bool const update_now =
+        update.do_update and (newton_iterations % update.update_every_newton_iter == 0);
 
       // solve linear problem
-      linear_iterations_last =
-        linear_solver.solve(increment, residual, update.do_update && threshold_exceeded);
+      linear_iterations_last = linear_solver.solve(increment, residual, update_now);
```

Comment:
Now you don't need the temporal `linear_iterations_last` anymore.

Reply:
yes, thanks, at least not as a class variable.